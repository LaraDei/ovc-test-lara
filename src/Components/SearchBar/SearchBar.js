import React, {Component} from 'react';
import './SearchBar.css';


export default class SearchBar extends Component{    

    render(){
        return(
            <div className="search-container">
                <form>
                <input type="text" placeholder="Search.." name="search" id="searchInput" onChange={(e)=>this.props.setTerm(e)}/>
                <button type="clear" onClick={this.props.clearTerm}>Clear</button>
                </form>
            </div>
        )
    }

}