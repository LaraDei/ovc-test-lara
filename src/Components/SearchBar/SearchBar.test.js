import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SearchBar from "./SearchBar";
// import {shallow } from 'enzyme';

describe("SearchBar component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SearchBar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders by default", () => {
    const wrapper = renderer.create(<SearchBar />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("it executes function onChange", () => {});

  it("it clears form with button is pushed", () => {});
});
