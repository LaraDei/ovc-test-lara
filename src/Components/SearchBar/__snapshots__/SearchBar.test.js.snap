// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders by default 1`] = `
<div
  className="search-container"
>
  <form>
    <input
      id="searchInput"
      name="search"
      onChange={[Function]}
      placeholder="Search.."
      type="text"
    />
    <button
      type="clear"
    >
      Clear
    </button>
  </form>
</div>
`;
